lib/galaxy/dependencies/pinned-requirements.txt

```diff
@@ -158,17 +168,22 @@ refgenconf==0.12.2
 regex==2024.11.6
 repoze-lru==0.7
 requests==2.32.3
+requests-cache==1.2.1
 requests-oauthlib==2.0.0
 requests-toolbelt==1.0.0
 requests-unixsocket==0.4.1
 rich==13.9.4
 rich-argparse==1.7.0
+rich-click==1.8.8
+roc-validator==0.4.2 ; python_full_version < '3.9.20'
+roc-validator==0.4.6 ; python_full_version >= '3.9.20' and python_full_version < '3.10'
+roc-validator==0.6.3 ; python_full_version >= '3.10'
```

Comment:
Maybe @nsoranzo or someone else can confirm, but **I think you can just keep the `roc-validator` lines and discard all the rest dependency updates**. Those will happen in the next automatic dependency update, and we can deal with potential linter changes then.

Thank you for working on this! :pray:

Reply:
Unfortunately there's a number of new dependencies that come in with roc-validator, so it's not that simple.
To minimise changes, you can:

```shell
$ git restore -s upstream/dev lib/galaxy/dependencies/
$ ./lib/galaxy/dependencies/update.sh -p roc-validator
```

This should revert all the changes to `lib/galaxy/dependencies/pinned-lint-requirements.txt` and some of the changes in the other requirements files.

Reply:
Related question: If this roc-validator is going to be used only in testing, it should be a dev-dependency, right? :thinking: